cpp/src/core/radsend.cpp

```diff
@@ -1957,15 +1957,15 @@ void radTSend::OutRelaxResultsInfo(double* RelaxStatusParamArray, int lenRelaxSt
 //#ifdef __JAVA__
 #if defined __JAVA__ || defined ALPHA__DLL__ || defined ALPHA__LIB__
 	int TotOutElem = lenRelaxStatusParamArray + 1;
-	double *TotOutArray = new double[TotOutElem];
-	if(TotOutArray == 0) { ErrorMessage("Radia::Error900"); return;}
-	double *t = TotOutArray;
+	std::vector<double> TotOutArray;
+	TotOutArray.resize(TotOutElem);
+	double *t = TotOutArray.data();
```

Comment:
Nice catch, and very good hint how to work with vector<double> data via double* !
In this case, double* RelaxStatusParamArray can probably be directly submitted to MultiDimArrayOfDouble(..); I'll check this further and correct. Thanks!

Reply:
Glad to be of help.

If you want to give ownership of an object when calling a function, C++11 makes it possible to give objects like a vector to a function without expensive copying. For example:

```cpp
void StoreMultiDimArrayOfDouble(std::vector<double> Array, int* Dims, int NumDims);

// And then call the function like this
StoreMultiDimArrayOfDouble(std::move(param), dims, n_dims);
```

Compared to passing a raw pointer with the soft contract that the ownership is given to the function, with this approach there is no ambiguity and much lower risk of mistakes.

If you want to go a little deeper, effectively the same thing could be done, ever so slightly more efficiently, with

```cpp
void StoreMultiDimArrayOfDouble(std::vector<double>&& Array, int* Dims, int NumDims);
```
(This is an "rvalue reference", a new kind of reference that is similar to a normal reference but it also means that the recipient of it is allowed to consume the object, the caller promises that the next method call is either the destructor or assignment operator.)